src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -187,9 +187,13 @@ public sealed partial class TokenCache : ITokenCacheInternal
                         requestParams.RequestContext.ApiEvent.DurationInCacheInMs += sw.ElapsedMilliseconds;
                     }
 
-                    // Don't cache PoP access tokens from broker
-                    if (msalAccessTokenCacheItem != null && !(response.TokenSource == TokenSource.Broker && response.TokenType == Constants.PoPAuthHeaderPrefix))
-                    {
+                    // Don't cache access tokens from broker
+#if iOS
```

Comment:
nit: better to place this complicated `if` condition in a private function, `ShouldPersistAccessToken()`